src/metabase/task/delete_inactive_tables.clj

```diff
@@ -0,0 +1,54 @@
+(ns metabase.task.delete-inactive-tables
+  (:require
+   [clojurewerkz.quartzite.jobs :as jobs]
+   [clojurewerkz.quartzite.schedule.cron :as cron]
+   [clojurewerkz.quartzite.triggers :as triggers]
+   [java-time :as t]
+   [metabase.config :as config]
+   [metabase.models.permissions :as perms]
+   [metabase.models.task-history :as task-history]
+   [metabase.task :as task]
+   [metabase.util.i18n :refer [trs]]
+   [metabase.util.log :as log]
+   [toucan2.core :as t2]))
+
+(set! *warn-on-reflection* true)
+
+(def ^:private inactive-table-max-days
+  "The maximum number of days to keep inactive tables around for."
+  (or (config/config-int :inactive-table-max-days) 180))
+
+(def ^:private job-key     "metabase.task.delete-inactive-tables.job")
+(def ^:private trigger-key "metabase.task.delete-inactive-tables.trigger")
+
+(defn- delete-inactive-tables!
+  []
+  (log/debug (trs "Cleaning up Inactive Tables older than {0} days" inactive-table-max-days))
+  (task-history/with-task-history {:task "delete-inactive-tables"}
+    (t2/with-transaction [_conn]
+      (let [to-delete-tables (t2/select [:model/Table :id :db_id :schema]
+                                        :active false
+                                        :updated_at [:< (t/minus (t/zoned-date-time) (t/seconds inactive-table-max-days))])]
```

Comment:
```suggestion
                                        :active     false
                                        :updated_at [:< (t/minus (t/zoned-date-time)
                                                                                    (t/days inactive-table-max-days))])]
```

The whitespace is me being fussy, but the `seconds` -> `days` seems like an essential bug?


```
just-now
#t "2023-11-07T12:44:42.878378088Z[UTC]"

> (t/minus just-now (t/seconds 180))
#t "2023-11-07T12:41:42.878378088Z[UTC]"

> (t/minus just-now (t/days 180))
#t "2023-05-11T12:44:42.878378088Z[UTC]"
```

Reply:
ha it was not caught by the test because the new but inactive table in the test has `updated_at` is `now`, so it's not qualified as less than 180 seconds.

I've updated the test to have 2 tables:
- one have updated_at less than 1 day from the threshold
- one have updated_at 1 day older than the threshold

[dcc27b0](https://github.com/metabase/metabase/pull/35443/commits/dcc27b0ac75d7f99c76c0d16e2008fc4d9ed6ea4)